kore/src/Kore/Syntax/Json/Internal.hs

```diff
@@ -307,25 +330,305 @@ fromPatternF (_ :< patt) = case patt of
+          where
+            sort = fromSort variableSort
+        TermLike.EndiannessF (Const endianness)
+            | Kore.Symbol{symbolConstructor, symbolParams} <- Endianness.toSymbol endianness ->
+                KJApp
+                    { name = fromKoreId symbolConstructor
+                    , sorts = map fromSort symbolParams
+                    , args = []
+                    }
+        TermLike.SignednessF (Const signedness)
+            | Kore.Symbol{symbolConstructor, symbolParams} <- Signedness.toSymbol signedness ->
+                undefined
+                    KJApp
+                        { name = fromKoreId symbolConstructor
+                        , sorts = map fromSort symbolParams
+                        , args = []
+                        }
+        TermLike.InjF inj
+            | ( Kore.Application
+                    Kore.Symbol{symbolConstructor, symbolParams}
+                    args
+                ) <-
+                Inj.toApplication inj ->
+                KJApp
+                    { name = fromKoreId symbolConstructor
+                    , sorts = map fromSort symbolParams
+                    , args
+                    }
+        TermLike.InternalBoolF (Const (InternalBool boolSort boolValue)) ->
+            encodeInternalValue boolSort $
+                if boolValue then "true" else "false"
+        TermLike.InternalBytesF (Const (InternalBytes bytesSort bytesValue)) ->
+            encodeInternalValue bytesSort $
+                Encoding.decode8Bit $
+                    ByteString.fromShort bytesValue
+        TermLike.InternalIntF (Const (InternalInt intSort intValue)) ->
+            encodeInternalValue intSort $
+                Text.pack $
+                    show intValue
+        TermLike.InternalStringF (Const (InternalString stringSort stringValue)) ->
+            encodeInternalValue stringSort stringValue
+        TermLike.InternalListF internalList -> encodeInternalList internalList
+        TermLike.InternalMapF internalMap -> encodeInternalAc internalMap
+        TermLike.InternalSetF internalSet -> encodeInternalAc internalSet
+      where
+        encodeInternalValue domainValueSort value =
+            KJDV
+                { sort = fromSort domainValueSort
+                , value
+                }
+
+        -- The encoded lists and ac types are all of the following form:
+        -- If the structure has no elements, we simply get the application of the unit symbol with no arguments
+        -- If the structure has exactly one child, we just get the child.
+        -- This child will be an application of internalElement symbol to its arguments
+        -- For multiple children we will get a \right-assoc{concatSymbol}(x1,...x_n)
+        foldApplication Kore.Symbol{symbolConstructor = unitSymbolConstructor, symbolParams = unitSymbolParams} Kore.Symbol{symbolConstructor = concatSymbolConstructor, symbolParams = concatSymbolParams} = foldAux
```

Comment:
For reference: https://github.com/runtimeverification/haskell-backend/blob/master/kore/src/Kore/Internal/TermLike/TermLike.hs#L1385-L1440 differs in `foldApplication` only.
Unfortunately there is no good way to avoid the duplication.